lib/finch.ex

```diff
@@ -57,6 +57,17 @@ defmodule Finch do
       used to configure proxying, https settings, or connect timeouts.
       """,
       default: []
+    ],
+    pool_max_idle_time: [
```

Comment:
I think this name makes sense, but adding it makes the existing `:max_idle_time` option seem a bit vague. We should probably deprecate that option in favor of a new `:conn_max_idle_time`. What do you think?